test/e2e/framework/util.go

```diff
@@ -3884,15 +3885,29 @@ func GetSigner(provider string) (ssh.Signer, error) {
 		// Otherwise revert to home dir
 		keyfile = "kube_aws_rsa"
 	case "vagrant":
-		keyfile := os.Getenv("VAGRANT_SSH_KEY")
+		keyfile = os.Getenv("VAGRANT_SSH_KEY")
 		if len(keyfile) != 0 {
 			return sshutil.MakePrivateKeySignerFromFile(keyfile)
 		}
 		return nil, fmt.Errorf("VAGRANT_SSH_KEY env variable should be provided")
+	case "local":
+		keyfile = os.Getenv("LOCAL_SSH_KEY") // maybe?
+		if len(keyfile) == 0 {
+			keyfile = "id_rsa"
+		}
+		// make sure keyfile exists on this host
+		key = filepath.Join(keydir, keyfile)
+		_, _, err := RunCmd("ls", key)
```

Comment:
Agreed. `ioutil.ReadFile(key)` is called by `MakePrivateKeySignerFromFile`. Removed the `ls` call.